server/src/main/java/org/apache/druid/server/coordinator/DruidCompactionConfig.java

```diff
@@ -126,7 +128,36 @@ public boolean isUseAutoScaleSlots()
   @JsonProperty
   public CompactionEngine getEngine()
   {
-    return compactionEngine;
+    return engine;
+  }
+
+
+  // Null-safe getters not used for serialization
+  public ClusterCompactionConfig clusterConfig()
+  {
+    return new ClusterCompactionConfig(
+        compactionTaskSlotRatio,
+        maxCompactionTaskSlots,
+        useAutoScaleSlots,
+        engine
+    );
+  }
+
+  public Map<String, DataSourceCompactionConfig> dataSourceToCompactionConfig()
```

Comment:
The name seems to suggest conversion when it actually returns a map.
```suggestion
  public Map<String, DataSourceCompactionConfig> dataSourceToCompactionConfigMap()
```